### PR TITLE
Reload command

### PR DESCRIPTION
**Pull request by astei**
I'm picking up where #120 leaves off. I would like to thank @mikroskeem for his initial work.

Features:

- [X] Supports the `/velocity reload` command
- [X] Most settings automatically update upon reload
- [X] Most proxy functionality still works on reload
- [X] Reload event for plugins
- [x] Changing/removing servers "in-flight"
- [x] Query stuff
- [x] A way to preserve existing user connections while closing the old port

### CODE REVIEW

File: api/src/main/java/com/velocitypowered/api/event/proxy/ProxyReloadEvent.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package com.velocitypowered.api.event.proxy;
+
+/**
+ * This event is fired when the proxy is reloaded by the user using {@code /velocity reload}.
+ */
+public class ProxyReloadEvent {
+
+  @Override
+  public String toString() {
+    return "ProxyInitializeEvent";
```
Comment (supertassu) [inline api/src/main/java/com/velocitypowered/api/event/proxy/ProxyReloadEvent.java]:
```suggestion
    return "ProxyReloadEvent";
```

File: proxy/src/main/java/com/velocitypowered/proxy/VelocityServer.java [hunk 2]
```diff
@@ -232,6 +234,37 @@ public boolean isShutdown() {
...
+
+    // Re-register servers
+    for (Map.Entry<String, String> entry : newConfiguration.getServers().entrySet()) {
+      ServerInfo newInfo =
+          new ServerInfo(entry.getKey(), AddressUtil.parseAddress(entry.getValue()));
+      Optional<RegisteredServer> rs = servers.getServer(entry.getKey());
+      if (!rs.isPresent()) {
+        servers.register(newInfo);
+      } else if (!rs.get().getServerInfo().equals(newInfo)) {
+        throw new IllegalStateException("Unable to replace servers in flight!");
```
[the remark this answers is not in the document]
Comment (mikroskeem) [reply proxy/src/main/java/com/velocitypowered/proxy/VelocityServer.java]: +1

To fallback server preferably?

File: proxy/src/main/java/com/velocitypowered/proxy/network/ConnectionManager.java [hunk 3]
```diff
@@ -90,8 +94,15 @@ public Bootstrap createWorker() {
             this.server.getConfiguration().getConnectTimeout());
   }
 
+  public void shutdown(InetSocketAddress oldBind) {
```
Comment (kashike) [inline proxy/src/main/java/com/velocitypowered/proxy/network/ConnectionManager.java]: `close` instead

File: proxy/src/main/java/com/velocitypowered/proxy/VelocityServer.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (astei) [reply proxy/src/main/java/com/velocitypowered/proxy/VelocityServer.java]: Was already planned.

**Pull request merged**